compiler/optimizations/gpuTransforms.cpp

```diff
@@ -482,7 +485,11 @@ void GpuKernel::populateBody(CForLoop *loop, FnSymbol *outlinedFunction) {
     std::vector<SymExpr*> symExprsInBody;
     collectSymExprs(node, symExprsInBody);
 
-    if (DefExpr* def = toDefExpr(node)) {
+    CallExpr* call = toCallExpr(node);
+    if(call && call->isPrimitive(PRIM_GPU_SET_BLOCKSIZE)) {
+        blockSize_ = toSymExpr(call->get(1));
+    }
```

Comment:
Current implementation looks like we're letting this primitive to exist in the kernel through codegen, where the codegen for it is basically a noop. Can we prevent that? Some options:

- We `continue` here.
- We set `copyNode=false` here. (My preference)

Note that whatever we choose to do, we also need to remove the primitive from the actual loop's body.

Reply:
Also note that making this primitive a `PRIMITIVE_R`, we *have to* remove it from the AST here. So that change, and this one complements one another.

Reply:
Fixed in 9d559a8ac3b726ecaf63d09af50ae5ddac2eeafc (I took the copyNode=false approach)